src/server/multi_command_squasher.cc

```diff
@@ -216,6 +222,15 @@ bool MultiCommandSquasher::ExecuteSquashed(facade::RedisReplyBuilder* rb) {
   if (order_.empty())
     return true;
 
+  // Multi non atomic does not lock ahead. So it's safe to preempt while we haven't
+  // really started the transaction.
+  // This is not true for `multi/exec` which uses `Execute()` but locks ahead before it
+  // calls `ScheduleSingleHop` below.
+  // TODO Investigate what are the side effects for allowing it `lock ahead` mode.
+  if (opts_.is_mult_non_atomic) {
```

Comment:
I dont remember if we discussed this -  Why do you need to throttle here and not use the same mecanism we have today for pipeline backpressure to throttle also if we have squashing_current_reply_size above limit